aptos-move/framework/aptos-token-objects/sources/aptos_token.move

```diff
@@ -382,48 +380,43 @@ module token_objects::aptos_token {
         key: String,
         type: String,
         value: vector<u8>,
-    ) acquires AptosToken {
+    ) acquires AptosCollection, AptosToken {
         let aptos_token = authorized_borrow(&token, signer::address_of(creator));
         assert!(
-            option::is_some(&aptos_token.property_mutator_ref),
+            are_properties_mutable(token),
             error::permission_denied(EPROPERTIES_NOT_MUTABLE),
         );
 
-        property_map::add(
-            option::borrow(&aptos_token.property_mutator_ref),
-            key,
-            type,
-            value,
-        );
+        property_map::add(&aptos_token.property_mutator_ref, key, type, value);
     }
 
     public fun add_typed_property<T: key, V: drop>(
         creator: &signer,
         token: Object<T>,
         key: String,
         value: V,
-    ) acquires AptosToken {
+    ) acquires AptosCollection, AptosToken {
         let aptos_token = authorized_borrow(&token, signer::address_of(creator));
         assert!(
-            option::is_some(&aptos_token.property_mutator_ref),
+            are_properties_mutable(token),
```

Comment:
What's the rationale behind this change? Basically you use a `bool` of a collection to decide whether tokens in this collection has properties mutable and store a mutable_ref in all tokens? Why do you have this ref. My guess is for potential upgrade to control each one individually but not sure how this can work later.

Reply:
I figure it is okay to just make this a collection level configuration for our no-code solution... the code complexity here isn't great and it is mostly just a bunch of control flags.. so I became a little oppressive here and there.